### PR TITLE
Rehaul of Makefile to enable easier dependency upgrades

### PR DESCRIPTION
**Pull request by pixelsandpointers**
Update of the dependency toolchain with the following changes:

1. Specify bare minimum dependencies in Pipfile
2. Run `make upgrade` to remove requirement files and lock file. This will further create new requirement files for each core and development dependencies. For compatibility, the script also removes the first line that pipenv creates.
3. If you want to install the package for development reasons, you can run `make install`, or `make` (if you wish to upgrade and install in one go).

Generally, use the Pipfile for dependency housekeeping and pip for installing the package.

### CODE REVIEW

**Comment by tomwallis** [comment]
As discussed directly, works for me after removing local editable in pipfile.

Consider also updating `.github/workflows/pytest.yml` to run `make`?

**Comment by pixelsandpointers** [comment]
So now that we have decided against the Pipfile, I would also remove the Makefile, Pipfiles, and the requirements, because the building already works as is and we have no dependency toolchain anymore.

This also requires no adjustment of the Github actions.

**Comment by pixelsandpointers** [comment]
@tomwallis please note that I had to change GH action to install development dependencies via `pip install -e ".[dev]"`.

File: pyproject.toml [hunk 1]
```diff
@@ -6,20 +6,35 @@ authors = [
...
+  'arviz'
+]
+
+[project.optional-dependencies]
+dev = [
+  'pytest',
+  'black',
+  'jupyterlab',
+  'ipykernel',
+  'graphviz',
```
Comment (tomwallis) [inline pyproject.toml]: I'm not sure whether to treat `graphviz` as a dev dependency or a main dependency. It's required if one wants to render the model using the `render` method. What do you think?

**Review by tomwallis — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (pixelsandpointers) [reply pyproject.toml]: We need it to visualize the pyro models, so it makes sense to add it as a dependency. Let's add it.

Comment (tomwallis) [reply pyproject.toml]: LGTM

**Pull request merged**